lib/queue.js

```diff
@@ -276,6 +276,37 @@ class Queue extends Emitter {
     });
   }
 
+  _getJobsByIds(ids) {
+    // If ids is a Set (i.e. returned from Queue._scanForJobs), convert it to an
+    // Array.
+    const idsArray = Array.from(ids);
+    if (!idsArray.length) {
+      return Promise.resolve([]);
+    }
+
+    // We need to re-ensure the queue is commandable, as we might be shutting
+    // down during this operation.
+    return this._commandable().then((client) => {
+      const got = helpers.deferred();
+      const commandArgs = [this.toKey('jobs')].concat(idsArray, got.defer());
```

Comment:
In the previous getJobs we would just use the job instance from the local jobs map if available; here it seems we fetch all jobs from redis and make new instances. Could we continue to use the local instances that are available and only `hmget` the missing ones?